lib/spack/spack/build_environment.py

```diff
@@ -188,6 +190,29 @@ def __call__(self, *args, **kwargs):
         return super().__call__(*args, **kwargs)
 
 
+class UndeclaredDependencyError(spack.error.SpackError):
+    """Raised if a dependency is invoking an executable through a module global, without
+    declaring a dependency on it.
+    """
+
+
+class DeprecatedExecutable:
+    def __init__(self, pkg: str, exe: str, exe_pkg: str) -> None:
+        self.pkg = pkg
+        self.exe = exe
+        self.exe_pkg = exe_pkg
+
+    def __call__(self, *args, **kwargs):
+        long_message = f"Is {self.pkg} missing a dependency on {self.exe_pkg}?"
```

Comment:
i would just keep it on one line. the second doesn't say anything interesting.

Reply:
Fixed